modules/core/04-channel/keeper/packet.go

```diff
@@ -197,9 +197,20 @@ func (k Keeper) RecvPacket(
 		return errorsmod.Wrap(err, "couldn't verify counterparty packet commitment")
 	}
 
+	// REPLAY PROTECTION: The recvStartSequence will prevent historical proofs from allowing replay
+	// attacks on packets processed in previous lifecycles of a channel. After a successful channel
+	// upgrade all packets under the recvStartSequence will have been processed and thus should be
+	// rejected.
+	recvStartSequence, _ := k.GetRecvStartSequence(ctx, packet.GetDestPort(), packet.GetDestChannel())
+	if packet.GetSequence() < recvStartSequence {
+		return errorsmod.Wrap(types.ErrPacketReceived, "packet already processed in previous channel upgrade")
```

Comment:
Does `previous version` sound a bit better here?

```suggestion
		return errorsmod.Wrap(types.ErrPacketReceived, "packet already processed in a previous channel version")
```

Reply:
I wanted to stay away from "version" here since you can upgrade a channel without changing its version